internal/pkg/api/handleEnroll.go

```diff
@@ -175,7 +175,7 @@ func (et *EnrollerT) _enroll(
 			ID:      agentID,
 			Version: ver,
 		},
-		Tags: req.Metadata.Tags,
+		Tags: removeDuplicateStr(req.Metadata.Tags),
```

Comment:
You may also want to consider reusing [`str.Set`](https://github.com/elastic/elastic-agent-libs/blob/f17cb1de6337603f16629d4754bcf8c27e0cc8be/str/stringset.go#L71) from the `elastic-agent-libs` module.  Normally I don't think we'd want to depend on another library _just_ for such simple functionality but since we are [already importing `elastic-agent-libs`](https://github.com/elastic/fleet-server/blob/fe0ec1e969018df23bb8d577aa91490324637722/go.mod#L10), I think it might make sense to reuse functionality from it here.

Reply:
Thanks for this suggestion @ycombinator - I see we can reduce the helper function to a one-liner using
```
return str.MakeSet(strSlice...).ToSlice()
```
but this would have the side effect of sorting the tags alphabetically, thus changing the current behaviour. I am not sure this would be desirable. Unless you had another suggestion in mind?

Reply:
Good point re: sorting and how that would change the existing behavior.  Reading through the [documentation on tags](https://www.elastic.co/guide/en/fleet/current/filter-agent-list-by-tags.html) and the [original issue that requested the tags feature](https://github.com/elastic/elastic-agent/issues/149), it seems that tags are agent metadata and order is not significant.  

But it'll be good to get confirmation about this from someone more experienced with the project.  @michel-laterman @blakerouse, WDYT? (both of you reviewed the [original PR](https://github.com/elastic/elastic-agent/pull/336) introducing tags during Agent enrollment)

Reply:
I don't think the order matters for tags

Reply:
👍 If the order doesn't matter, then I can change the implementation to the one above.